### DISCUSSION TITLE
Preserve order of the selected items

### DISCUSSION THREAD

**Discussion by ghost** [orchidjs/tom-select#48] [Q&A] [answered]
I have a select like this for specifying a film's genres.

```
<select name="genres[]" multiple placeholder="Select one or more" autocomplete="off">
  <option value="1">Action</option>
  <option value="2">Adventure</option>
  <option value="3">Comedy</option>
  <option value="4">Horror</option>
  <option value="5">Thriller</option>
  <option value="6">Western</option>
</select>
```

```
new TomSelect("select[name='genres[]']", {
    onItemAdd:function(){
        this.setTextboxValue('');
        this.refreshOptions();
    },
    create: false,
});
```

A film can belong to multiple genres, and the order in which they're specified is important.

But if I select "_Western_" and then "_Adventure_", the actual values submitted are "_Adventure_" and "_Western_", so the specified order is lost.

How should I go about preserving these values (while still having the dropdown sorted alphabetically)?
I should also be able to prefill the select with items in a particular order.

Any help is appreciated, thank you! ✌️

**Comment by oyejorge** [comment, marked as answer]
Sorry, looks like this is actually a bug. I'll have a fix in a little bit

**Reply by ghost** [reply]
Thank you! I was wondering, even if the order of the selected items is preserved when reading from TomSelect (after fixing the bug), how would one go about specifying an order when initialising the plugin with a couple of items already selected? 

E.g. if I select "Horror" and "Adventure" (in this order) I end up (after saving, reloading and reinitialising the plugin) with the following select:

```
<select name="genres[]" multiple>
  <option value="1">Action</option>
  <option value="2" selected>Adventure</option>
  <option value="3">Comedy</option>
  <option value="4" selected>Horror</option>
  <option value="5">Thriller</option>
</select>
```

Naturally TomSelect would read those options and auto select "Adventure", then "Horror", which however is not the intended order. Is there a way of working around this?

**Reply by oyejorge** [reply]
Yeah two ways to approach that.

A: Order the &lt;option> tags
```html
<select id="create-here" name="genres[]" multiple>
  <option value="4" selected>Horror</option>
  <option value="2" selected>Adventure</option>
  <option value="1">Action</option>
  <option value="3">Comedy</option>
  <option value="5">Thriller</option>
</select>
```

B: In settings
```js
new TomSelect("#create-here",{items:[4,2]});
```

**Reply by ghost** [reply]
Option B works great. Also I confirm the order of items is preserved after the fix in v1.4.2, thank you. ✌️

**Reply by andreasmatuszewski** [reply]
@oyejorge Are you looking forward to fix this bug? Because way B is no option to me. Select2 does not reorder the option, but i dont want to use it..

On .sync() it randomly reorders the options.
